src/string_pool.cr

```diff
@@ -86,6 +86,23 @@ class StringPool
     get slice.to_unsafe, slice.size
   end
 
+  # Returns a `String` with the contents of the given *slice*, or `nil` if no
+  # such string exists in the pool.
+  #
+  # ```
+  # require "string_pool"
+  #
+  # pool = StringPool.new
+  # bytes = "abc".to_slice
+  # pool.get?(bytes) # => nil
+  # pool.empty?      # => true
+  # pool.get(bytes)  # => "abc"
+  # pool.empty?      # => false
```

Comment:
**suggestion**: the doc could have a second call to `get?` that would now return a String.